### PR TITLE
UGENE-7217 SAFE_POINT "Object relation is not valid!" in "In Silico PCR" sample

### PR DESCRIPTION
**Pull request by rasputinkirill**
Most of the XML tests from 'cmdline/pcr' failed because of this issue.
So, we do not need additional tests.

### CODE REVIEW

File: src/plugins/pcr/src/ExtractProductTask.cpp [hunk 1]
```diff
@@ -240,7 +241,11 @@ void ExtractProductTask::run() {
     AnnotationTableObject *annotations = new AnnotationTableObject(productSequence.getName() + " features", dbiRef);
     annotations->addAnnotations(QList<SharedAnnotationData>() << getPrimerAnnotation(product.forwardPrimerMatchLength, U2Strand::Direct, productSequence.length()));
     annotations->addAnnotations(QList<SharedAnnotationData>() << getPrimerAnnotation(product.reversePrimerMatchLength, U2Strand::Complementary, productSequence.length()));
-    annotations->addObjectRelation(GObjectRelation(GObjectReference(sequenceObject), ObjectRole_Sequence));
+    GObjectReference ref(sequenceObject);
```
Comment (yalgaer) [inline src/plugins/pcr/src/ExtractProductTask.cpp:244]: Why adding a random non-existing URL  here helps to solve the problem?

Comment (rasputinkirill) [reply src/plugins/pcr/src/ExtractProductTask.cpp:244]: Impossible to add object relation with empty url in GObjectReference. 
So we add temporary url for object to pass this.
As i understand this situation is only possible when this task started through WD.

Comment (DmitriiSukhomlinov) [reply src/plugins/pcr/src/ExtractProductTask.cpp:244]: Actually, it looks like a crutch, but I don't have a better offer

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Review by smlnkcv — APPROVED** [review]

File: src/plugins/pcr/src/ExtractProductTask.cpp (hunk 1 above)
Comment (yalgaer) [reply src/plugins/pcr/src/ExtractProductTask.cpp:244]: A long story short: hacks like this won't go.

If GObjectReference requires an URL the URL must be real. Why can't we re-use the URL of the document for the sequence object?

**Review by yalgaer — CHANGES_REQUESTED** [review]

**Comment by rasputinkirill** [comment]
This document has an empty URL.
Because this is workflow and its URL will be defined in "Write Sequence" worker.

**Pull request closed without merging**